### ISSUE TITLE
TestNodeMetadata fails on Go 1.11.1+

### ISSUE THREAD

**Issue by nmittler** [istio/istio#9994]
1.11 works (which is what CI uses), but the method fails on both 1.11.1 and 1.11.2 with:

```bash
go test ./pkg/bootstrap/ -run TestNodeMetadata
2018-11-15T22:53:00.507177Z	warn Env variable istio.io/enable [{"abc": 20}] failed json unmarshal: json: cannot unmarshal number into Go value of type string
--- FAIL: TestNodeMetadata (0.00s)
   bootstrap_config_test.go:372: envs => [ISTIO_METAJSON_LABELS={"istio":"sidecar","l1":"v1","l2":"v2"} ISTIO_METAJSON_ANNOTATIONS={"istio.io/enable":"{\"abc\": 20}"}]
       nm=> map[istio:sidecar l1:v1 l2:v2 istio.io/enable:{"abc": 20}]
   bootstrap_config_test.go:382: Maps are not equal.
       got: map[l1:v1 l2:v2 istio.io/enable:{"abc": 20} abc: istio:sidecar]
       want: map[l2:v2 istio:sidecar istio.io/enable:{"abc": 20} l1:v1]
FAIL
FAIL	istio.io/istio/pkg/bootstrap 0.104s
```
It appears that there is a difference in behavior of the built-in json library.

**Comment by mandarjog** [comment]
@hzxuzhonghu the original test included the type of strings that we expect to see in the field.
This fixes the test failure, but it is going to fail when this sort of data is present in a deployment.
We need to look at this closely, can you investigate?

**Comment by hzxuzhonghu** [comment]
@mandarjog It's because the json lib changed behavior since golang 1.11.

https://play.golang.org/p/SqdmEDT2SZ2

golang 1.10  {\"abc\": 20}==>map[string]string{}

golang latest {\"abc\": 20}==>map[string]string{"abc":""}

**Comment by hzxuzhonghu** [comment]
xref: https://github.com/golang/go/issues/27318

**Comment by mandarjog** [comment]
We expect the strings like the one used in the test. Most keys in the stringified json should be strings. I want to make sure that the original code works. We have effectively disabled string test.

**Comment by smawson** [comment]
Status on this? What needs to happen here if anything?

**Comment by nmittler** [comment]
This was fixed by #10002. The github autoclose feature must not have like the PRs reference to this issue for some reason. Closing now.
